frontends/web/src/routes/device/bitbox02/passphrase.tsx

```diff
@@ -57,39 +57,45 @@ class Passphrase extends Component<Props, State> {
 
   public componentDidMount() {
     getDeviceInfo(this.props.deviceID)
-      .then(({ mnemonicPassphraseEnabled }) => this.setState({
-        // before enabling/disabling we show 1 or more pages to inform about the feature
-        // each page has a continue button that jumps to the next or finally toggles passphrase
-        // infoStep counts down in decreasing order
-        infoStep: mnemonicPassphraseEnabled
-          ? INFO_STEPS_DISABLE
-          : INFO_STEPS_ENABLE,
-        passphraseEnabled: mnemonicPassphraseEnabled,
-      }))
-      .catch(error => {
-        console.error(error);
-        alertUser(this.props.t('genericError'));
-      });
+      .then(result => {
+        if (!result.success) {
+          console.error(result.message);
+          alertUser(this.props.t('genericError'));
+          return;
+        }
+        const { mnemonicPassphraseEnabled } = result.deviceInfo;
+        this.setState({
+          // before enabling/disabling we show 1 or more pages to inform about the feature
+          // each page has a continue button that jumps to the next or finally toggles passphrase
+          // infoStep counts down in decreasing order
+          infoStep: mnemonicPassphraseEnabled
+            ? INFO_STEPS_DISABLE
+            : INFO_STEPS_ENABLE,
+          passphraseEnabled: mnemonicPassphraseEnabled,
+        });
+      })
+      .catch(console.error);
   }
 
-  private togglePassphrase = () => {
+  private togglePassphrase = async () => {
     const { deviceID, t } = this.props;
     const enable = !this.state.passphraseEnabled;
     this.setState({ status: 'progress' });
-    setMnemonicPassphraseEnabled(deviceID, enable)
-      .then(() => getDeviceInfo(deviceID))
-      .then(({ mnemonicPassphraseEnabled }) => {
-        this.setState({
-          passphraseEnabled: mnemonicPassphraseEnabled,
-          status: 'success',
-        });
-      })
-      .catch((e) => {
-        route(`/device/${deviceID}`);
-        alertUser(t(`passphrase.error.e${e.code}`, {
-          defaultValue: e.message || t('genericError'),
-        }));
+    const result = await setMnemonicPassphraseEnabled(deviceID, enable);
+    if (!result.success) {
+      route(`/device/${deviceID}`);
+      alertUser(t(`passphrase.error.e${result.code}`, {
+        defaultValue: result.message || t('genericError'),
+      }));
+      return;
+    }
+    const deviceInfoResult = await getDeviceInfo(deviceID);
+    if (deviceInfoResult.success) {
+      this.setState({
+        passphraseEnabled: deviceInfoResult.deviceInfo.mnemonicPassphraseEnabled,
+        status: 'success',
       });
+    }
```

Comment:
funny that alertUser still shows even if before route is called.. it's because the weird global alertUser.

having alertUser before route would at least make the intention clear, but it would still be a bit confusing I guess. keeping unless somebody has a strong opinion.

Reply:
In theory api calls such as setMnemonicPassphraseEnabled should never error, but if they would.. for example during development, it would only say uncaught promise rejection or similar... 
is better to always have in a try/catch and console.error ?

Reply:
> is better to always have in a try/catch and console.error ?

Tbh I have no strong opinion on this. Seems like our codebase isn't so consistent with it anyway (some are using try/catch, some dont). But catching the error and logging it would be okay, at least that way during development it's easier to see what error is occurring. Although not sure if that's really worth it.